packages/neon/neon/lib/src/pages/login.dart

```diff
@@ -128,129 +57,70 @@ class _LoginPageState extends State<LoginPage> {
-                                    ),
-                                    Text(AppLocalizations.of(context).loginWorksWith),
-                                    const SizedBox(
-                                      height: 20,
-                                    ),
-                                    const NextcloudLogo(),
-                                    Form(
-                                      key: _formKey,
-                                      child: TextFormField(
-                                        focusNode: _focusNode,
-                                        decoration: const InputDecoration(
-                                          hintText: 'https://...',
-                                        ),
-                                        keyboardType: TextInputType.url,
-                                        initialValue: widget.serverURL,
-                                        validator: (final input) => validateHttpUrl(context, input),
-                                        onFieldSubmitted: (final input) {
-                                          if (_formKey.currentState!.validate()) {
-                                            _loginBloc.setServerURL(input);
-                                          } else {
-                                            _focusNode.requestFocus();
-                                          }
-                                        },
-                                      ),
-                                    ),
-                                    NeonLinearProgressIndicator(
-                                      visible: serverConnectionStateSnapshot.data == ServerConnectionState.loading,
-                                    ),
-                                    if (serverConnectionStateSnapshot.data == ServerConnectionState.unreachable) ...[
-                                      NeonException(
-                                        AppLocalizations.of(context).errorUnableToReachServer,
-                                        onRetry: _loginBloc.refresh,
-                                      ),
-                                    ],
-                                    if (serverConnectionStateSnapshot.data ==
-                                        ServerConnectionState.maintenanceMode) ...[
-                                      NeonException(
-                                        AppLocalizations.of(context).errorServerInMaintenanceMode,
-                                        onRetry: _loginBloc.refresh,
-                                      ),
-                                    ],
-                                  ],
-                                ),
+          child: Scaffold(
+            resizeToAvoidBottomInset: true,
+            appBar: AppBar(
+              leading: accountsSnapshot.data?.isNotEmpty ?? false
+                  ? BackButton(
+                      onPressed: () {
```

Comment:
Oh and maybe it's worth not using a back Icon but rather use an IconButton with an `X` to show that this cancels the login rather than going back a page in the loginflow.

There should be a widget called `CloseButton` but I'd have to check

Reply:
The whole manual back button and listener can actually be removed. With the new routing system it works out of the box :)